lib/rules/require-curly-braces.js

```diff
@@ -97,14 +91,12 @@ module.exports.prototype = {
 
         if (typeIndex['if'] || typeIndex['else']) {
             file.iterateNodesByType('IfStatement', function(node) {
-                if (typeIndex.if && node.consequent.type !== 'BlockStatement') {
+                if (typeIndex['if'] && isNotABlockStatement(node.consequent)) {
                     addError('If', node);
                 }
-                if (typeIndex['else'] && node.alternate &&
-                    node.alternate.type !== 'BlockStatement' &&
-                    node.alternate.type !== 'IfStatement'
-                ) {
-                    addError('Else', node);
+                if (typeIndex['else'] && isNotABlockStatement(node.alternate) &&
+                    node.alternate.type !== 'IfStatement') {
```

Comment:
Yes, it needs to be kept only for the require case, because otherwise `else if...` would need to be `else { if...` because technically that else doesn't have curly braces.